python_modules/libraries/dagster-databricks/dagster_databricks/databricks_pyspark_step_launcher.py

```diff
@@ -325,6 +326,18 @@ def log_compute_logs(self, log: DagsterLogManager, run_id: str, step_key: str) -
                 f" {step_key}. Check the databricks console for more info."
             )
 
+    def log_run_page_url(self, context: StepExecutionContext, databricks_run_id: int):
+        run = self.databricks_runner.client.workspace_client.jobs.get_run(databricks_run_id).as_dict()
+        run_page_url = run.get("run_page_url")
+        yield DagsterEvent.from_step(
+            event_type=DagsterEventType.ENGINE_EVENT,
+            step_context=context,
+            event_specific_data=EngineEventData(
+                marker_start="Databricks Job",
+                metadata={"step_key": context.node_handle.name, "databricks_run_url": MetadataValue.url(run_page_url)}
```

Comment:
I would also add a check for the case that the url is None (and just not set any metadata in that case), just to avoid erroring if something changes.

Reply:
> any particular reason for including the step key here? this is already populated in a separate field on the DagsterEvent and rendered in the UI in a different column

That's a good point! I have removed the redundant logging of `step_key`.

Reply:
> it turns out adding a message arg to DagsterEvent.from_step per comment below removes the step key from the event message

Oh that's a weird interaction -- I guess it's fine to keep the step key in the metadata in that case, not a huge deal.

> Is there a preferred way to check this case in code?

I was thinking it could just be something like:

```python
metadata = {"step_key": ...}
run_page_url = ...
if run_page_url is not None:
    metadata["run_page_url"] = MetadataValue.url(run_page_url)
```

just because if the URL happens to be blank, I'd rather it doesn't show up at all rather than have a broken-seeming link (good to know that it doesn't error if None is passed in though)

Reply:
That's a good point. It could be useful to have some indication of an issue from the run being submitted (e.g. hit an api rate limit) and I thought an empty link could spur that? I suppose the dagster run would eventually failover if that were the case 🤔 . Wdyt?

Reply:
I think it would be hard for a random user to connect a missing URL in that metadata box to "there's an issue that needs solving" (I think that sort of prompting / alerting is better left for more targeted solutions) -- I was imagining it'd be more likely for that to become empty in some imagined future where they stop populating that field in the API response or something like that.

I don't feel super strongly, as it seems unlikely that this thing ever actually will be blank, and I feel better knowing that it won't cause an error if that is the case.

Reply:
Yeah, that makes sense. Let's exclude it as you suggested 👍 

Here is an example of how it renders.
![Screenshot 2024-02-28 at 5 37 12 PM](https://github.com/dagster-io/dagster/assets/23409221/065e2245-48c0-4bbf-8222-d1503331afe6)